### ISSUE TITLE
Audio player crashes after about 30 minutes of continuous operation

### ISSUE THREAD

**Issue by SadisticMystic** [joel16/3DShell#43]
When I leave the audio player running, it always seems to cause a Luma crash after about half an hour, with the lid constantly open. This happens regardless of shuffle or sequential play mode, and has happened at both track transitions and in the middle of a song. Running on o3DS, here's the most recent crash dump.

[crash_dump_00000004.dmp.zip](https://github.com/joel16/3DShell/files/2780394/crash_dump_00000004.dmp.zip)

**Comment by joel16** [comment]
I'm going to be refactoring a lot of the audio code in the near future to allow things like seeking, getting progress, time etc. Once it's available I'll let you know so that you can test it for me and let me know if this issue still persists.

**Comment by chrismslist** [comment]
Same I also get this issue on the newest version :(

**Comment by chrismslist** [comment]
Also what's the status on the next update?

**Comment by joel16** [comment]
I haven't had the time to work on this yet. Probably not until some of my exams clear up.

**Comment by chrismslist** [comment]
Alright cool, good to know it's still in development

**Comment by joel16** [comment]
Ok, so you can access the new changes in the nighlty builds (you can update within the app). At the moment it crashes if you try exiting the app or pressing the home button only after using next/previous buttons or if the music player automatically plays the next audio file after the current one is finished. However if you play one audio file, and exit out of it using "B" those crashes won't happen. 

The update also adds support for the following formats: **IT, MOD, S3M and XM** (Opus and Flac have been fixed now as well)

And it displays the current/total duration of each audio file as well as a small progress bar.

If you need to revert, you'll have to re-download the latest version from the releases tab.

**Comment by chrismslist** [comment]
Cool epic 👍

**Comment by joel16** [comment]
Thanks but that doesn't exactly answer the question. Can any of you guys confirm the issue with the 30 mins of continuous playback exists in the latest nightly?

**Comment by SadisticMystic** [comment]
I tried running the audio player in standard operation on the new version for over an hour without incident, so that seems to be fine. However, switching on the Shuffle button led to problems, starting with the progress bar getting stuck at the end of a track and failing to advance to the next one (it still responded to the "skip track button", though I had to keep pressing the button again to advance each track thereafter), and eventually to this crash.

[crash_dump_00000006.dmp.zip](https://github.com/joel16/3DShell/files/3133497/crash_dump_00000006.dmp.zip)

The repeat button, meanwhile, seems to be fine as well.
